### DISCUSSION TITLE
Synchronizing takes a long time

### DISCUSSION THREAD

**Discussion by npodonnell** [romanz/electrs#357] [Q&A] [answered]
Hello,

I've set up electrs together with bitcoind on my home server. When I point my Electrum instance to the electrs instance, it seems to take a long time to sync.

Software versions are:

bitcoind: v0.21.0
electrs: v0.8.7
electrum: v4.0.9

I built electrs from source. I'm running bitcoind without the `txindex` option (0)

Electrs unit file:
```
# electrs-btc.unit
[Unit]
Description=Electrs Bitcoin
After=network.target

[Service]
ExecStart=/usr/local/bin/electrs -vvvv --network bitcoin --daemon-dir /var/lib/bitcoind --blocks-dir /var/lib/bitcoind/blocks --db-dir /electrs-btc/db --electrum-rpc-addr "0.0.0.0:50001" --cookie-file /var/lib/bitcoind/.cookie
WorkingDirectory=/electrs-btc
User=bitcoin
Group=bitcoin
Type=simple
KillMode=process
TimeoutSec=60
Restart=always
RestartSec=60

[Install]
WantedBy=multi-user.target
```

Electrum command:

```
~/bin/electrum-4.0.9-x86_64.AppImage --oneserver --server=asus:50001:t
```

`asus` is my home server.

Edit: It seems to be synched now, but often will go back into a synching state and stay synching for a long time (like hours)

My question is - is there any thing I can do with my setup here to improve the synching time?

And yes - my bitcoind instance is fully synched :)

**Comment by Kixunil** [comment]
I suggest checking `top` and `free` to determine if you have enough memory etc. If there's no problem, it could be a bug.

**Reply by npodonnell** [reply]
I think my server (which is actually an old laptop), may be just too slow.

Output from top:

```
top - 11:23:33 up  1:30,  2 users,  load average: 4.60, 3.52, 1.96
Tasks: 232 total,   3 running, 228 sleeping,   0 stopped,   1 zombie
%Cpu(s): 47.4 us, 18.4 sy, 14.2 ni, 16.0 id,  3.9 wa,  0.0 hi,  0.1 si,  0.0 st
MiB Mem :   5825.4 total,    162.3 free,   3017.2 used,   2645.9 buff/cache
MiB Swap:    976.0 total,    715.7 free,    260.3 used.   2434.7 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                                                            
   4234 user      20   0   12008   4576   4136 R  98.8   0.1   0:33.73 pk-debconf-help                                                                                                                                                                    
    949 bitcoin   20   0 2528564   1.1g   4960 S  96.5  18.7  24:07.14 electrs                                                                                                                                                                            
   2145 user      20   0 3823944 207744  85232 S  19.4   3.5   0:53.00 gnome-shell                                                                                                                                                                        
      1 root      20   0  188396  10196   5808 R  18.7   0.2   0:11.85 systemd                                                                                                                                                                            
   1143 bitcoin   20   0 4295832   1.1g 211432 S   4.4  19.8   9:52.13 bitcoind                                                                                                                                                                           
    852 message+  20   0   10824   5984   3596 S   4.1   0.1   0:05.04 dbus-daemon                                                                                                                                                                        
   1994 user      20   0  779616  53728  29372 S   2.8   0.9   0:09.66 Xorg         
```

Electrs has very heavy CPU usage. This is a somewhat crappy, old machine, so maybe it's just not beefy enough. I'm also running Ubuntu 20.10 Desktop, so there's quite a lot of GUI stuff hogging the CPU/Memory. I'll try switching off all the Xorg/GUI stuff and see if that helps.


Edit: When I close electrum, the electrs CPU usage goes back down to ~2%. My wallet is a few years old and has quite a lot of active addresses and several hundred transactions in the history. I think this is also a factor.

**Reply by Kixunil** [reply]
The current version of `electrs` is known to not perform great with large wallets. There's a possible solution, it just needs some work. I was actually looking at it today a bit but got distracted.

**Reply by npodonnell** [reply]
Yes, it seems that way. It ran slightly better with the `--jsonrpc-import` rather than reading the `.blk` files directly. But it's still syncing quite a lot.

**Reply by romanz** [reply]
Just from curiosity, could you please try the latest `p2p` branch on your machine?
It would require re-indexing the blocks, but it should (hopefully) take less resources than the current `master` branch.
Please see https://github.com/romanz/electrs/issues/308#issuecomment-806061229 for more details.

**Comment by romanz** [comment, marked as answer]
Please try the latest [0.9.1 release](https://github.com/romanz/electrs/releases/tag/v0.9.1) :)
